dask/array/_shuffle.py

```diff
@@ -294,8 +294,10 @@ def _getitem(obj, index):
 
 
 def concatenate_arrays(arrs, sorter, axis):
-    concatenate = concatenate_lookup.dispatch(type(arrs[0]))
-    return np.take(concatenate(arrs, axis=axis), np.argsort(sorter[1]), axis=axis)
+    typ = type(arrs[0])
+    take = take_lookup.dispatch(typ)
+    concatenate = concatenate_lookup.dispatch(typ)
+    return take(concatenate(arrs, axis=axis), np.argsort(sorter[1]), axis=axis)
```

Comment:
```suggestion
    return take_lookup(
        concatenate_lookup(arrs, axis=axis),
        np.argsort(sorter[1]),
        axis=axis,
    )
```

Reply:
@rjzamora these are lists so the `__call__` will not work for `concatenate_lookup`

Reply:
but will do for `take_lookup`

Reply:
Ah, sorry - My mistake. I was probably thinking of the [dataframe `concat` function](https://github.com/dask/dask/blob/091f4e9bd226836faeaf30bdc3b5182ba14d22a6/dask/dataframe/dispatch.py#L34) - We could add a similar function to dask.array, but I think the way you did things is fine.